### DISCUSSION TITLE
Extend existing bundle?

### DISCUSSION THREAD

**Discussion by someguynamedjosh** [bevyengine/bevy#1217] [Q&A]
I have a situation where I want to spawn an entity that is a sprite with a couple of extra components. Right now I am doing it like this:
```rust
#[derive(Bundle)]
struct BuildingBundle {
    // Begin sprite bundle...
    pub sprite: Sprite,
    pub mesh: Handle<Mesh>, // TODO: maybe abstract this out
    pub material: Handle<ColorMaterial>,
    pub main_pass: MainPass,
    pub draw: Draw,
    pub visible: Visible,
    pub render_pipelines: RenderPipelines,
    pub transform: Transform,
    pub global_transform: GlobalTransform,
    // End sprite bundle...
    pub pos: BuildingPos,
}
```
Is there a way to use the existing definition of `SpriteBundle` and then add `BuildingPos` to it?

**Comment by someguynamedjosh** [comment]
I could also do something like this:
```rust
#[derive(Bundle)]
struct BuildingBundle {
    pub pos: BuildingPos,
}

fn building_bundles(
    ...
) -> (SpriteBundle, BuildingBundle) {
    ...
}

let (a, b) = building_bundles(...);
commands.spawn(a).with_bundle(b);
```

If I added a custom trait I could give commands a function like `spawn_with_bundles` that would accept a tuple of bundles and expand it accordingly. I think that would be ergonomic enough but it would be nicer if there was a built-in solution.

**Comment by CleanCut** [comment]
> Is there a way to use the existing definition of `SpriteBundle` and then add `BuildingPos` to it?

From that question, it seems like you're looking to organize logic, not data, so maybe instead of looking to custom data structures, consider using a helper function instead?

```rust
fn spawn_building(commands: &mut Commands, ...) {
    ...
    commands.spawn(SpriteBundle {
            material: materials.add(texture_handle.into()),
            ..Default::default()
        })
        .with(BuildingPos { ... });
}
```

Does that seem like what you were looking for?

**Comment by alice-i-cecile** [comment]
See #792 and follow-up discussion :) The bundle nesting tech described in #1570 should solve your specific need here.
